lib/manageiq/api/client/collection.rb

```diff
@@ -42,6 +43,58 @@ def self.subclass(name)
             const_set(klass_name, klass)
           end
         end
+
+        def ar_search(options)
+          search(query_parameters_from_ar_query_options(options))
+        end
+
+        private
+
+        def query_parameters_from_ar_query_options(options)
+          api_params = {}
+          [:offset, :limit].each { |opt| api_params[opt] = options[opt] if options[opt] }
+          api_params[:attributes] = options[:select].join(",") if options[:select].present?
+          if options[:where]
+            api_params[:filter] ||= []
+            conditional_parameter_from_ar_query_options(api_params[:filter], "=", options[:where])
+          end
+          if options[:not]
+            api_params[:filter] ||= []
+            conditional_parameter_from_ar_query_options(api_params[:filter], "!=", options[:not])
+          end
+          if options[:order]
+            order_parameters_from_ar_query_options(api_params, options[:order])
+          end
+          api_params
+        end
+
+        def conditional_parameter_from_ar_query_options(filter_param, condition, ar_option)
+          ar_option.each do |attr, values|
+            Array(values).each do |value|
+              value = "'#{value}'" if value.kind_of?(String) && !value.match(/^(NULL|nil)$/i)
+              filter_param << "#{attr}#{condition}#{value}"
+            end
+          end
+        end
+
+        def order_parameters_from_ar_query_options(api_params, ar_option)
+          api_params[:sort_by] = []
+          api_params[:sort_order] = []
+          ar_option.each do |sort_attr, sort_order|
+            api_params[:sort_by] << sort_attr
+            case sort_order
+            when /^asc/i
+              sort_order = "asc"
```

Comment:
Might as well inline these with the `whens` so it's not as visually verbose ... you can also move the `sort_order =` outside of the case...

```ruby
sort_order =
  case sort_order
  when /^asc/i  then "asc"
  when /^desc/i then "desc"
  else raise "Invalid sort order #{sort_order} specified for attribute #{sort_attr}"
  end